pkg/operator/configobserver/featuregates/observe_featuregates.go

```diff
@@ -95,3 +82,47 @@ func (f *featureFlags) ObserveFeatureFlags(genericListers configobserver.Listers
 
 	return observedConfig, errs
 }
+
+func (f *featureFlags) getFeatureNames(fg *configv1.FeatureGate) ([]string, error) {
+	newConfigValue := []string{}
+	formatEnabledFunc := func(fs string) string {
+		return fmt.Sprintf("%s=true", fs)
+	}
+	formatDisabledFunc := func(fs string) string {
+		return fmt.Sprintf("%s=false", fs)
+	}
+
+	if fg.Spec.FeatureSet == configv1.CustomNoUpgrade {
+		if fg.Spec.FeatureGateSelection.CustomNoUpgrade == nil {
+			return nil, fmt.Errorf(".spec.featureGateSelection.CustomNoUpgrade is empty but the flag %s was set", configv1.CustomNoUpgrade)
```

Comment:
We don't want to fail.  This function should on fail on fatal issues.  You can add this to the validating admission plugin.